dashboard/modules/job/job_manager.py

```diff
@@ -288,18 +284,93 @@ class JobManager:
     # Time that we will sleep while tailing logs if no new log line is
     # available.
     LOG_TAIL_SLEEP_S = 1
+    JOB_MONITOR_LOOP_PERIOD_S = 1
 
     def __init__(self):
         self._status_client = JobStatusStorageClient()
         self._log_client = JobLogStorageClient()
         self._supervisor_actor_cls = ray.remote(JobSupervisor)
 
+        self._running_jobs: Dict[str, JobStatusInfo] = dict()
+        self._recover_running_jobs()
+
+    def _recover_running_jobs(self):
+        """Recovers all running jobs from the status client.
+
+        Each will be added to self._running_jobs and reconciled.
+        """
+        all_jobs = self._status_client.get_all_jobs()
+        for job_id, status_info in all_jobs.items():
+            if not status_info.status.is_terminal():
+                self._running_jobs[job_id] = status_info
```

Comment:
When a job moves into a terminal state, where does it get deleted from `self._running_jobs`?

Reply:
I'm actually realizing that we don't need `_running_jobs` at all, let me remove it.

Reply:
updated